File: backend/app/api/v1/schemas/manual_strength.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, StrictInt, StrictStr


class ManualStrengthSessionCreateRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    started_at: datetime
    timezone_name: StrictStr
    duration_minutes: StrictInt
    body_regions: list[StrictStr]
    perceived_exertion: StrictInt | None = None
    notes: StrictStr | None = None


class ManualStrengthSessionUpdateRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    started_at: datetime | None = None
    timezone_name: StrictStr | None = None
    duration_minutes: StrictInt | None = None
    body_regions: list[StrictStr] | None = None
    perceived_exertion: StrictInt | None = None
    notes: StrictStr | None = None


class ManualStrengthTrainingLoadResponse(BaseModel):
    load_value: float
    method: str
    unit: str
    quality: str
    warnings: list[str]
    algorithm_version: str
    calculated_at: datetime


class ManualStrengthSessionResponse(BaseModel):
    id: UUID
    started_at: datetime
    timezone_name: str
    duration_minutes: int
    body_regions: list[str]
    perceived_exertion: int | None
    notes: str | None
    created_at: datetime
    updated_at: datetime
    training_load: ManualStrengthTrainingLoadResponse

File: frontend/src/types/manualStrength.ts
export type BodyRegion = "full_body"|"chest"|"back"|"shoulders"|"arms"|"core"|"quadriceps"|"hamstrings"|"glutes"|"calves";
export interface ManualStrengthTrainingLoad {load_value:number;method:"strength_rpe"|"strength_duration";unit:"points";quality:"low"|"medium";warnings:string[];algorithm_version:string;calculated_at:string}
export interface ManualStrengthSession {id:string;started_at:string;timezone_name:string;duration_minutes:number;body_regions:BodyRegion[];perceived_exertion:number|null;notes:string|null;created_at:string;updated_at:string;training_load:ManualStrengthTrainingLoad}
export interface ManualStrengthSessionCreate {started_at:string;timezone_name:string;duration_minutes:number;body_regions:BodyRegion[];perceived_exertion:number|null;notes:string|null}
export type ManualStrengthSessionUpdate=Partial<ManualStrengthSessionCreate>;
